.github/workflows/checkstyle.yml

```
@@ -0,0 +1,56 @@
+      - '/*.java'  # Monitor all Java files
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+    permissions: # Add permissions for check enforcement
+      contents: read
+      pull-requests: write
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)